packages/block-editor/src/components/inserter/tabs.native.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { Animated, Dimensions, View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useCallback,
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { SegmentedControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import BlocksTypesTab from './blocks-types-tab';
+import ReusableBlocksTab from './reusable-blocks-tab';
+import styles from './style.scss';
+
+const TAB_ANIMATION_DURATION = 250;
+
+function InserterTabs( {
+	listProps,
+	onSelect,
+	rootClientId,
+	showReusableBlocks,
+	tabIndex,
+} ) {
+	const [ windowWidth, setWindowWidth ] = useState(
+		Dimensions.get( 'window' ).width
+	);
+	const tabAnimation = useRef( new Animated.Value( 0 ) ).current;
+	const lastScrollEvents = useRef( [] ).current;
+
+	useEffect( () => {
+		Dimensions.addEventListener( 'change', onDimensionsChange );
+		return () =>
+			Dimensions.removeEventListener( 'change', onDimensionsChange );
+	}, [] );
+
+	useEffect( () => {
+		Animated.timing( tabAnimation, {
+			duration: TAB_ANIMATION_DURATION,
+			toValue: tabIndex,
+			useNativeDriver: true,
+		} ).start();
+
+		// Notify upstream with the last scroll event of the current tab.
+		const lastScrollEvent = lastScrollEvents[ tabIndex ];
+		if ( lastScrollEvent ) {
+			listProps.onScroll( { nativeEvent: lastScrollEvent } );
+		}
```

Comment:
The bottom-sheet component uses the `onScroll` handler to [calculate the value of `bounces`](https://github.com/WordPress/gutenberg/blob/2a938548e8d2fed8a25b30db2b8675236089f176/packages/components/src/mobile/bottom-sheet/index.native.js#L295-L301), this value is also used to control whether the modal should enable the swipe interaction that closes it, so it's important to explicitly call `onScroll` when the current tab changes to prevent undesired effects.

_Related to this [comment](https://github.com/WordPress/gutenberg/pull/28495/files#r627322782)._